tasks/docker-engine.yml

```diff
@@ -19,74 +19,8 @@
     repo: 'deb https://apt.dockerproject.org/repo ubuntu-{{ ansible_distribution_release }} main'
     update_cache: yes
 
-- name: see if docker is already installed
-  stat: path=/usr/bin/docker get_checksum=False
-  register: docker_exec_stat_result
-  always_run: True 
-
-- name: Determine installed version
-  command: docker --version
-  changed_when: False
-  always_run: True
-  register: docker_version_result
-  when: "{{ docker_exec_stat_result.stat.exists }}"
-
-# "docker --version" output is formatted like this:
-#
-#     Docker version 1.10.2, build c3959b1
-#
-- name: store installed docker version
-  set_fact:
-    installed_docker_version: "{{ docker_version_result.stdout
-                                  | regex_replace('^Docker version ', '')
-                                  | regex_replace(', build [0-9a-f]+$', '') }}"
-  when: "{{ docker_exec_stat_result.stat.exists }}"
-
-- name: collect container data from before upgrade
-  collect_container_configs:
-  register: collect_container_configs_result
-  when: "{{ docker_exec_stat_result.stat.exists
-            and docker_attempt_upgrade_fixes }}"
-  always_run: True
-
-# It should be OK to just run the apt install of the new docker package, regardless of the
-# state of docker. However, there's an issue that sometimes occurs in that case, where the
-# processes for the containers that existed before the upgrade aren't stopped by the upgrade, but
-# instead are left hanging around with parent PID 1. Shutting down docker before the upgrade seems
-# to prevent this.
-- name: shut down docker before upgrade
-  service: name=docker state=stopped
-  when: "{{ docker_exec_stat_result.stat.exists and installed_docker_version != docker_version }}"
- 
 - name: Install Docker Engine
   apt:
     name: "docker-engine={{ docker_version }}-0~{{ ansible_distribution_release }}"
```

Comment:
probably don't really need to change, but i've seen some packages that have a different debian revision on different ubuntu releases (so trusty has one debian revision and percise has another) and i think we probably want the latest version for the release.

It is probably a rare case, but it is possible. not sure we want to parameterize this... but might save time if it ever comes up.

https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-Version
https://docs.docker.com/engine/installation/linux/ubuntulinux/#/update-your-apt-sources